workflow/snakemake_rules/main_workflow.smk

```diff
@@ -1394,6 +1394,15 @@ rule build_description:
         with open(output.description, "w", encoding = "utf-8") as o:
             o.write(template.safe_substitute(context))
 
+def rule_exists(name):
+    # Snakemake 7.32.1 changed the way rules are stored internally
+    # https://github.com/snakemake/snakemake/commit/65c79a48f956077839bb5ab1ea8d60a5f0ddecab
+    from packaging import version # an explicit dependency of augur
+    if version.parse(snakemake.__version__) < version.parse('7.32.1'):
+        return name in rules.__dict__
+    else:
+        return name in rules.__dict__['_rules']
```

Comment:
I'd recommend avoiding the private internals and instead defining this in terms of accessibility via the public API:

```python
def rule_exists(name):
    try:
        return bool(getattr(rules, name, None))
    except:
        return False
```

This works on both versions.

Reply:
Sure. Done.